packages/core/src/internal/controllers/button-anchor.controller.ts

```diff
@@ -0,0 +1,37 @@
+import { ReactiveController, ReactiveElement } from 'lit';
+
+export type ButtonAnchor = ReactiveElement &
+  HTMLElement & {
+    readonly: boolean;
+  };
+
+/**
+ * Shim for allowing custom element button types to be wrapped with anchors as Safari prevents use of extending native element types.
+ * Will detect presence of anchor and set component to a readonly button state.
+ *
+ * Example:
+ *
+ * ```html
+ * <a href="#">
+ *  <cds-button></cds-button>
+ * </a>
+ * ```
+ */
+export function buttonAnchor<T extends ButtonAnchor>(): ClassDecorator {
+  return (target: any) => target.addInitializer((instance: T) => new ButtonAnchorController(instance));
+}
+
+export class ButtonAnchorController<T extends ButtonAnchor> implements ReactiveController {
+  constructor(private host: T) {
+    this.host.addController(this);
+  }
+
+  hostUpdated() {
+    const hasAnchor = this.host.parentElement?.tagName === 'A';
+    if (hasAnchor && this.host.parentElement) {
```

Comment:
Wouldn't the `?.` part of `this.host.parentElement?.tagName` on the previous line handle the `this.host.parentElement` check for us?

Reply:
Yeah I thought so as well, but the `parentElement` property on `HTMLElement` technically can be null on elements that were created but not yet appended to the DOM. So this means a required null check by TS.